internal/provider/kubernetes/routes.go

```diff
@@ -34,7 +34,7 @@ func (r *gatewayAPIReconciler) processTLSRoutes(ctx context.Context, gatewayName
 	}
 
 	for _, tlsRoute := range tlsRouteList.Items {
-		tlsRoute := tlsRoute
+		tlsRoute := tlsRoute //nolint:copyloopvar
```

Comment:
curious why its needed here ?

Reply:
same as https://github.com/envoyproxy/gateway/pull/4047#discussion_r1716918702, in L95 it's referenced and got appended into `resourceTree.TLSRoutes`.